Sources/Instrumentation/Tracing/Span.swift

```diff
@@ -0,0 +1,94 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Tracing open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Tracing project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Baggage
+import Dispatch
+
+public protocol Span {
+    var operationName: String { get }
+
+    var startTimestamp: DispatchTime { get }
+    var endTimestamp: DispatchTime? { get }
+
+    var baggage: BaggageContext { get }
+
+    var events: [SpanEvent] { get }
+    mutating func addEvent(_ event: SpanEvent)
+
+    var attributes: [String: SpanAttribute] { get }
+    subscript(attributeName attributeName: String) -> SpanAttribute? { get set }
+
+    var onEnd: (Span) -> Void { get }
+    mutating func end(at timestamp: DispatchTime)
```

Comment:
Would you be ok with `finish()` instead?

Reply:
Yep, `finish` is good as well. I followed the OT spec on this one. https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#end

Reply:
Hmm... I see, well that's the spec I guess. Maybe just a `TODO: that we may consider finish?`
End does not sound like a "verb" which this kind of feels it should be 🤔

Reply:
Good point, I'll add the TODO

```suggestion
    // TODO: naming is defined in the spec, but we may want to consider finish instead as it sounds more like a verb
    mutating func end(at timestamp: DispatchTime)
```